packages/api/src/beacon/routes/lightclient.ts

```diff
@@ -119,12 +131,44 @@ export function getDefinitions(_config: ChainForkConfig): RouteDefinitions<Endpo
             }
             return value;
           },
+          serialize: (data, meta) => {
+            const chunks: Uint8Array[] = [];
+            for (const [i, update] of data.entries()) {
+              const version = meta.versions[i];
+              const forkDigest = cachedBeaconConfig().forkName2ForkDigest(version);
+              const serialized = getLightClientForkTypes(version).LightClientUpdate.serialize(update);
+              const length = ssz.UintNum64.serialize(4 + serialized.length);
+              chunks.push(length, forkDigest, serialized);
+            }
+            return Buffer.concat(chunks);
+          },
+          deserialize: (data) => {
+            let offset = 0;
+            const updates: LightClientUpdate[] = [];
+            while (offset < data.length) {
+              const length = ssz.UintNum64.deserialize(data.subarray(offset, offset + 8));
+              const forkDigest = ssz.ForkDigest.deserialize(data.subarray(offset + 8, offset + 12));
+              const version = cachedBeaconConfig().forkDigest2ForkName(forkDigest);
+              updates.push(
+                getLightClientForkTypes(version).LightClientUpdate.deserialize(
+                  data.subarray(offset + 12, offset + 8 + length)
+                )
+              );
+              offset += 8 + length;
+            }
+            return updates;
+          },
         },
         meta: {
           toJson: (meta) => meta,
           fromJson: (val) => val as {versions: ForkName[]},
-          toHeadersObject: () => ({}),
-          fromHeaders: () => ({versions: []}),
+          toHeadersObject: (meta) => ({
+            [MetaHeader.Version]: meta.versions.join(","),
+          }),
+          fromHeaders: (headers) => {
+            const versions = headers.getOrDefault(MetaHeader.Version, "");
+            return {versions: versions === "" ? [] : (versions.split(",") as ForkName[])};
```

Comment:
Other clients won't be setting the header which means the metadata (`res.meta()`) will be empty, however this is not a big issue since it's not required to deserialize the response.